### PR TITLE
Fix more types

### PR DESCRIPTION
**Pull request by brandonkelly**
Continuation of #19332. Caught a few more missing `null`s.

| Q             | A
| ------------- | ---
| Is bugfix?    | ❌
| New feature?  | ❌
| Breaks BC?    | ❌

### CODE REVIEW

**Review by samdark — APPROVED** [review]

**Review by Arhell — APPROVED** [review]

File: framework/filters/AccessRule.php [hunk 1]
```diff
@@ -26,16 +26,16 @@
 class AccessRule extends Component
 {
     /**
-     * @var bool whether this is an 'allow' rule or 'deny' rule.
+     * @var bool|null whether this is an 'allow' rule or 'deny' rule.
```
[the remark this answers is not in the document]
Comment (brandonkelly) [reply framework/filters/AccessRule.php]: Yeah good call, done (023985ebd3539421f4fdc30f709ece7638cb5180). Also removed the `null` from `Validator::$skipOnEmpty` as there’s no special null handling for that (86d9c1706943a2f4f8cde2f67256d2d1bb2d5834).

**Comment by brandonkelly** [comment]
Going to tackle a couple other things, so just converted the PR to a draft.

**Comment by brandonkelly** [comment]
Alright… pretty happy with where it’s at for now.

File: framework/filters/AccessRule.php [hunk 2]
```diff
@@ -28,14 +28,14 @@ class AccessRule extends Component
     /**
      * @var bool whether this is an 'allow' rule or 'deny' rule.
      */
-    public $allow;
+    public $allow = true;
```
Comment (bizley) [inline framework/filters/AccessRule.php:31]: Shouldn't it be `false`? With `true` we are changing the default here.

Comment (brandonkelly) [reply framework/filters/AccessRule.php:31]: Doh! Yes. Fixed.

**Review by bizley — APPROVED** [review]

**Comment by bizley** [comment]
Thank you very much!

**Pull request merged**